datafusion/src/datasource/datasource.rs

```diff
@@ -104,15 +80,6 @@ pub trait TableProvider: Sync + Send {
         limit: Option<usize>,
     ) -> Result<Arc<dyn ExecutionPlan>>;
 
-    /// Returns the table Statistics
-    /// Statistics should be optional because not all data sources can provide statistics.
-    fn statistics(&self) -> Statistics;
-
-    /// Returns whether statistics provided are exact values or estimates
-    fn has_exact_statistics(&self) -> bool {
```

Comment:
this is also technically an API change (that `has_exact_statistics` is now encoded in the Statistics itself) -- I think it is a good change.

Reply:
correct! this change was flagged in the associated issue #962